### PR TITLE
Remove Launch button from editor top bar

### LINKED ISSUE #53109: Remove launch button in the page editor
Let's remove the launch button in the editor for now and have another round of design work to find a better placement for it.

Screenshot 2021-05-24 at 10 29 18

We have enough subjective evidence that the "launch" button in the editor header is confusing customers:

- Issues https://github.com/Automattic/wp-calypso/issues/52590 https://github.com/Automattic/wp-calypso/issues/52561 https://github.com/Automattic/wp-calypso/pull/41211
- Internal & user testing bring this up frequently (recent p1621458244077600-slack-dotcom-pod-design-selection and p58i-avT-p2)
- Layout breaks easily, even on bigger screens, especially with languages where texts are longer (https://github.com/Automattic/wp-calypso/issues/50758)

    Screenshot 2021-05-24 at 10 42 52

    Screenshot 2021-05-24 at 10 50 55

- Translating "publish" and "launch" has been challenging (p1604066018174600-slack-luna):

    Screenshot 2021-05-24 at 10 37 18

   The above has since been updated, but terms are still often close at quick glance and show how this might be challenging in other locales as well:

    Screenshot 2021-05-24 at 10 37 18

### Context

Initially, when designed & implemented (https://github.com/Automattic/wp-calypso/pull/39048), the "Publish" button was relabelled as "Save" because it was causing issues with React (https://github.com/Automattic/wp-calypso/pull/42508).

Screen Shot 2020-01-24 at 4 32 52 PM

Adjusting terminology is challenging because it's coming from the core (pbAok1-BD-p2), and needs to be updated in quite a few places to be consistent.

We also have few other areas where people can launch their sites:

- Frontend preview

    Screenshot 2021-05-24 at 10 52 20

- Settings (though this is hidden for sites with WP-Admin nav unification views enabled)

    Screenshot 2021-05-24 at 10 51 55

- Site setup checklist (excluding Atomic sites but that's being fixed, see https://github.com/Automattic/wp-calypso/issues/50523)

    Screenshot 2021-05-24 at 10 52 51

### Next

- Remove the launch button and resume regular functionality in the editor, and observe analytics and user testing.
- I know we have the "focused launch" flow starting from this point; the goal is to expand that flow to other places as well and not depend on this one button.
- Explore other placements for launch. Better design placement for this action might be within the FSE site editor. We might even have some earlier designs around.

### PR DESCRIPTION
**Pull request by razvanpapadopol**
#### Changes proposed in this Pull Request

* Remove Launch button from editor for all sites except AnchorFm.

#### Testing instructions

* Create a site with `/new`
* Do the same with `/start`
* Sandbox the two sites
* In both cases, when opening the editor, Launch button shouldn't be displayed as it is in production.

**Follow up**
- [x] Create an issue to handle proper code cleanup https://github.com/Automattic/wp-calypso/issues/53118

Fixes https://github.com/Automattic/wp-calypso/issues/53109

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php [hunk 1]
```diff
@@ -76,27 +76,14 @@ function enqueue_launch_button_script_and_style( $site_launch_options ) {
...
-		case 'gutenboarding-launch':
-			$script_name = 'gutenboarding-launch';
-			break;
-		case 'focused-launch':
-			$script_name = 'focused-launch';
-			break;
-		default:
-			// For redirect or invalid flows, skip & exit early.
-			return;
-	}
```
Comment (StefanNieuwenhuis) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:94]: This cleanup feels a bit like scope creep.

Comment (StefanNieuwenhuis) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:94]: This feels a bit like scope creep since it isn't necessary for removing the launch button.

Comment (StefanNieuwenhuis) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:94]: This feels a bit like scope creep since it isn't necessary for removing the launch button.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts [hunk 2]
```diff
@@ -75,18 +80,6 @@ domReady( () => {
...
-					// Save post in the background while focused launch flow opens
-					dispatch( 'automattic/launch' ).openFocusedLaunch();
-					delayedSavePost();
-					break;
-				case SITE_LAUNCH_FLOW:
-					// Save post first before redirecting to launch url
-					( async () => {
-						await savePost();
-						window.top.location.href = launchUrl;
-					} )();
```
Comment (StefanNieuwenhuis) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:88]: See previous comment about scope creep.

**Review by StefanNieuwenhuis — APPROVED** [review]
Tested as per testing instructions and everything works like a charm (see screenshots)! 
I left some comments about scope creep but I think we can leave it in this PR, since we're going to address cleanups in #53118 anyway.

### The editor toolbar for the site created with _/new_
Screenshot 2021-05-25 at 10 49 24

### The editor toolbar for the site created with _/start_
Screenshot 2021-05-25 at 10 53 05

### The editor toolbar for the site created with the AnchorFM flow
Screenshot 2021-05-25 at 10 55 01

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts (hunk 2 above)
Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:88]: I'm fine with keeping this but it's just dead code since the button is not there in these 2 cases. 
Also, I think it may lead to confusion to have these around because of the workaround we have in place for AnchorFM (see the `@TODO` above).

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php (hunk 1 above)
Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php:94]: While I have no strong opinion on https://github.com/Automattic/wp-calypso/pull/53115#discussion_r638623547, in this case we have an opportunity to decrease the bundle size immediately by not loading the `focused-launch` code for non-AnchorFM sites.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts (hunk 2 above)
Comment (alshakero) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:88]: Yeah, I agree that it's better to remove it.
